### PR TITLE
Bug fix/gharial improvement rename

### PR DESCRIPTION
**Pull request by hkernbach**
- Improved the rename c++ function in gharial (only used in general graph currently) 
- added test 

Reminder: 
This also needs a PR for 3.4, after review!

### CODE REVIEW

**Comment by hkernbach** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1041/

File: arangod/Graph/Graph.cpp [hunk 1]
```diff
@@ -550,6 +596,33 @@ bool Graph::hasOrphanCollection(std::string const& collectionName) const {
...
+      renamed = it.second.renameCollections(oldName, newName);
+    } else {
+      it.second.renameCollections(oldName, newName);
+    }
+  }
+
+  // orphans
+  if (hasOrphanCollection(oldName)) {
+    _orphanColls.erase(oldName);
+    _orphanColls.insert(oldName);
```
Comment (jsteemann) [inline arangod/Graph/Graph.cpp]: this makes no sense... first deleting the item and then inserting the same item again.

Comment (hkernbach) [reply arangod/Graph/Graph.cpp]: Thanks. Good you found it. Also there’s a test missing for that scenario. Will fix.

Comment (jsteemann) [reply arangod/Graph/Graph.cpp]: :+1:

**Comment by hkernbach** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/1975/

**Comment by hkernbach** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1120/

File: arangod/Graph/Graph.h [hunk 2]
```diff
@@ -150,10 +157,15 @@ class Graph {
...
   std::map<std::string, EdgeDefinition> const& edgeDefinitions() const;
 
+  /// @brief get the cids of all edgeCollections as reference
+  std::map<std::string, EdgeDefinition> & edgeDefinitions();
+
   bool hasEdgeCollection(std::string const& collectionName) const;
   bool hasVertexCollection(std::string const& collectionName) const;
   bool hasOrphanCollection(std::string const& collectionName) const;
 
+  bool renameCollections(std::string const& oldName, std::string const& newName);
```
Comment (jsteemann) [inline arangod/Graph/Graph.h]: rename (hihi) to `renameCollection`

**Comment by hkernbach** [comment]
@mchacki please check after jenkins job:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1542/

**Comment by hkernbach** [comment]
^-- I think it is now safe to merge.

**Pull request merged**